### PR TITLE
[XLA:CPU][oneDNN] Add post-ops for oneDNN Convolutions

### PR DESCRIPTION
**Pull request by akhilgoe**
This PR adds support for multiple post-ops for oneDNN Convolution and adds tests to verify functionality. New Post-ops supported:

1. Relu
2. Tanh
3. Gelu (Approx.)
4. Gelu (Exact)
5. Relu6
6. Sigmoid
7. Elu
8. Elementwise scalar product

### CODE REVIEW

File: xla/service/cpu/onednn_contraction_rewriter.cc [hunk 1]
```diff
@@ -196,6 +204,36 @@ std::optional<float> GetConstantValueAsFloat32(const HloInstruction* inst) {
   }
 }
 
+auto GetOneDnnContractionVariant(
```
Comment (akhilgoe) [inline xla/service/cpu/onednn_contraction_rewriter.cc:207]: Overlap with https://github.com/openxla/xla/pull/19099

File: xla/service/cpu/onednn_contraction_rewriter.h [hunk 2]
```diff
@@ -74,6 +76,26 @@ class OneDnnContractionRewriter : public HloModulePass {
           contraction, __VA_ARGS__);                                   \
   }
 
+using OneDnnContractionVariant =
```
Comment (akhilgoe) [inline xla/service/cpu/onednn_contraction_rewriter.h:79]: Overlap with https://github.com/openxla/xla/pull/19099

File: xla/service/cpu/onednn_convolution.h [hunk 3]
```diff
@@ -31,6 +31,7 @@ extern void __xla_cpu_runtime_OneDnnConvolution(void* result, void** args);
 template <>
 struct PrimitiveTrait<kOnednnConvConfig> {
   using pointer_type = xla::cpu::OneDnnConvolutionConfig*;
+  static const BackendConfigOneofCase kConfigVal = kOnednnConvConfig;
```
Comment (akhilgoe) [inline xla/service/cpu/onednn_convolution.h:34]: Overlap with https://github.com/openxla/xla/pull/19099

File: xla/service/cpu/onednn_matmul.h [hunk 4]
```diff
@@ -43,6 +43,7 @@ extern void __xla_cpu_runtime_OneDnnMatMulReorder(void* result, void** args);
 template <>
 struct PrimitiveTrait<kOnednnMatmulConfig> {
   using pointer_type = xla::cpu::OneDnnMatMulConfig*;
+  static const BackendConfigOneofCase kConfigVal = kOnednnMatmulConfig;
```
Comment (akhilgoe) [inline xla/service/cpu/onednn_matmul.h:46]: Overlap with https://github.com/openxla/xla/pull/19099

File: xla/service/cpu/onednn_util.h [hunk 5]
```diff
@@ -72,7 +72,7 @@ typedef BackendConfig::BackendConfigOneofCase BackendConfigOneofCase;
 template <typename PrimDesc>
 std::unique_ptr<PrimDesc> CreateOneDnnPrimDesc(HloInstruction*);
 
-template <BackendConfigOneofCase config>
+template <BackendConfigOneofCase config, typename TransformationType = void>
```
Comment (akhilgoe) [inline xla/service/cpu/onednn_util.h:75]: Overlap with https://github.com/openxla/xla/pull/19099

**Review by akhilgoe — COMMENTED** [review]
Hi @penpornk, this PR has minor overlap in some header files with https://github.com/openxla/xla/pull/18616. I have marked those areas for your convenience. Thanks!

**Review by penpornk — CHANGES_REQUESTED** [review]
@akhilgoe Sorry for the delay! https://github.com/openxla/xla/pull/18616 has been merged. Could you please help resolve the conflicts?

**Comment by akhilgoe** [comment]
Thanks @penpornk, I have resolved the conflicts.

**Review by penpornk — APPROVED** [review]
Thank you again for the PR!

**Pull request closed without merging**